amundsen_application/api/utils/search_utils.py

```diff
@@ -0,0 +1,54 @@
+from typing import Dict, List  # noqa: F401
+
+# These can move to a configuration when we have custom use cases outside of these default values
+valid_search_fields = {
+    'column',
+    'database',
+    'schema',
+    'table',
+    'tag'
+}
+
+
+# TODO: Consolidate with marshall_table_partial after schemas are unified & delete
+def map_table_result(result: Dict) -> Dict:
+    return {
+        'type': 'table',
+        'key': result.get('key', None),
+        'name': result.get('name', None),
+        'cluster': result.get('cluster', None),
+        'description': result.get('description', None),
+        'database': result.get('database', None),
+        'schema_name': result.get('schema_name', None),
+        'last_updated_epoch': result.get('last_updated_epoch', None),
+    }
+
+
+def generate_query_json(*, filters: Dict = {}, page_index: str, search_term: str) -> Dict:
```

Comment:
I'd say within the scope of this PR we can consolidate names across the new endpoints, but I think this might take more discussion than we have time to prioritize. My initial thoughts
1. `term` would be my preference for the search term
2. `page_index` (python) / `pageIndex` (typescript) [conversion should be done in api call] would actually be my preference for that because the search service code uses the concept of `index` to refer to the elastic search index to use (i.e. table index or user index). So I would want to use the same name across services as well
3. This task itself would touch a lot of files in a way that requires a lot of attention and disrupts other projects -- I think the best time to do it would be if we can get it in before dashboards work has to start touching redux/python code. Let me know what you think. Perhaps we can sync up offline to spec out dashboards work but also strategically plan 1 sprint of code cleanup that can get completed before dashboards is in its final stages.

Reply:
I'm fine with the names as long as they are consistent. I just don't want 3 names that refer to the same concept. 

I think doing this change just after https://github.com/lyft/amundsenfrontendlibrary/pull/382 but before publishing the 2.0.0 release would be ideal. I actually think `query` is better than `term` since it is more descriptive.

Reply:
I'll file a ticket. But given recent priorities with badge filtering I don't know id we'll get it before publishing the release. More likely it can at least be done before dashboards.